tests/test_connections/test_ledger/test_ledger.py

```diff
@@ -135,6 +138,61 @@ async def test_not_hanging(self) -> None:
         blocking_task.cancel()
 
 
+class TestRequestDispatcher:
+    """Test `RequestDispatcher` class."""
+
+    dispatcher: RequestDispatcher
+    loop: asyncio.AbstractEventLoop
+
+    def setup(self) -> None:
+        """Setup test vars."""
+        logger = logging.getLogger(type(self).__class__.__name__)
+        state = AsyncState(ConnectionStates.connected)
+        self.loop = asyncio.get_event_loop()
+        self.dispatcher = DummyRequestDispatcher(
+            logger=logger, connection_state=state, loop=self.loop
+        )
+
+    def dummy_func(self, sleep: float) -> bool:
+        """A dummy function that sleeps and returns True."""
+        time.sleep(sleep)
+        return True
+
+    async def dummy_async_func(self, sleep: float) -> bool:
+        """A dummy async function that sleeps and returns True."""
+        await asyncio.sleep(sleep)
+        return True
+
+    @pytest.mark.asyncio
+    async def test_wait_for_happy_path(self) -> None:
+        """Tests that wait_for works when timeout is bigger than execution time of callable."""
+        should_finish_in = 0.5  # 0.5 seconds
+        timeout = 1  # we give 2 the amount of time for it to finish
+
+        return_value = await self.dispatcher.wait_for(
+            lambda: self.dummy_func(should_finish_in), timeout=timeout
+        )
+        assert return_value, "dummy_func() should've returned True"
+
+    @pytest.mark.asyncio
+    async def test_wait_for_raise_excp(self) -> None:
+        """Tests that wait_for works when timeout is less than execution time of callable."""
+        should_finish_in = 0.5  # 0.5 seconds
+        timeout = 0.25  # we give 2 the amount of time for it to finish
```

Comment:
And how about something like this?

```suggestion
        # we increase the timeout by a tiny amount to test that a `TimeoutError` will be raised
        timeout = 0.25
        timeout_increase = 0.1
        should_finish_in = timeout + timeout_increase
```